pytui/main.py

```diff
@@ -1,19 +1,47 @@
+import subprocess  # noqa: S404
 import sys
 
-from backends.stackoverflow import StackOverflowFinder
-from display.display import Display
-from parsers.python import PythonParser
+from parse import findall
+from rich import print
+
+
+def parse_err(txt: str) -> tuple:
+    """Extract error message and packages invloved in stack trace.
+
+    The specific error message is assumed to be the last line that is not indented.
+    """
+    error_msg = ''
+    for line in txt.splitlines():
+        if line == line.strip():
+            error_msg = line.strip()
+
+    files = [
+        line['file']
+        for line
+        in findall('File "{file}", line {line_no}, in {method}', txt)
+    ]
+
+    packages = set()
+    key = 'site-packages'
+    for file in files:
+        if key not in file:
+            continue
+        path_elements = file.split('/')
+        packages.add(path_elements[path_elements.index(key) + 1])
+
+    return error_msg, packages
 
 
 def main() -> None:
     """Run the application"""
-    all_text = sys.stdin.read()
+    args = sys.argv
+    args[0] = 'python'  # Replace pytui with 'python' to act as a runner
+    result = subprocess.call(args, stderr=subprocess.PIPE, shell=False)  # noqa: S603
 
-    parser = PythonParser(all_text)
-    error_message = parser.parse_error()
-    backend = StackOverflowFinder()
-    search_results = backend.search(error_message)
-    Display(error_message, search_results).start()
+    if result.stderr:
```

Comment:
I'm getting this error:
```
(.venv) ➜  Perceptive-Porcupines-Code-Jam-8 git:(interface) pytui example/runs_with_error.py
0
1
2
Traceback (most recent call last):
  File "/Users/vagishvela/.pyenv/versions/3.9.5/bin/pytui", line 8, in <module>
    sys.exit(main())
  File "/Users/vagishvela/.pyenv/versions/3.9.5/lib/python3.9/site-packages/pytui/main.py", line 41, in main
    if result.stderr:
AttributeError: 'int' object has no attribute 'stderr'
```

if you `print(result)` it just returns a `1`. Maybe something is missing in the code or setup?

Reply:
`subprocess.call()` returns the error code of running the script. Maybe doing `subprocess.run()` would be better. https://docs.python.org/3/library/subprocess.html#subprocess.run

Reply:
Looks like this fixes the issue